components/experimental/src/dimension/units/format.rs

```diff
@@ -53,47 +53,52 @@ impl<'l> Writeable for FormattedUnit<'l> {
     }
 }
 
+impl_display_with_writeable!(FormattedUnit<'_>);
+
 #[test]
 fn test_basic() {
     use icu_locale_core::locale;
-    use writeable::Writeable;
+    use writeable::assert_writeable_eq;
 
     use crate::dimension::units::formatter::UnitsFormatter;
-    use crate::dimension::units::options::UnitsFormatterOptions;
-    use crate::dimension::units::options::Width;
-
-    let locale = locale!("en-US").into();
-    let meter = "meter";
-    let fmt = UnitsFormatter::try_new(&locale, meter, Default::default()).unwrap();
-    let value = "12345.67".parse().unwrap();
-    let formatted_unit = fmt.format_fixed_decimal(&value, meter);
-    let mut sink = String::new();
-    formatted_unit.write_to(&mut sink).unwrap();
-    assert_eq!(sink.as_str(), "12,345.67 m");
+    use crate::dimension::units::options::{UnitsFormatterOptions, Width};
 
-    let locale = locale!("de-DE").into();
-    let meter = "meter";
-    let fmt = UnitsFormatter::try_new(&locale, meter, Default::default()).unwrap();
-    let value = "12345.67".parse().unwrap();
-    let formatted_unit = fmt.format_fixed_decimal(&value, meter);
-    let mut sink = String::new();
-    formatted_unit.write_to(&mut sink).unwrap();
-    assert_eq!(sink.as_str(), "12.345,67 m");
+    let test_cases = [
+        (
+            locale!("en-US").into(),
```

Comment:
nit: move the `into`s into the loop

Reply:
Done here: https://github.com/unicode-org/icu4x/pull/5234